pkg/tdexdconnect/tdexdconnect.go

```diff
@@ -35,9 +36,13 @@ func Encode(
 
 // EncodeToString encodes the given args into a base64 string URL.
 func EncodeToString(
-	schema, rpcServerAddr string, certBytes, macBytes []byte,
+	scheme, rpcServerAddr string, certBytes, macBytes []byte,
 ) (string, error) {
-	u, err := Encode(schema, rpcServerAddr, certBytes, macBytes)
+	if certBytes != nil && scheme == "http" {
+		return "nil", errors.New("http protocol invalid with cert provided")
+	}
```

Comment:
this should be in `Encode` and also `Decode`.

Reply:
Done